sw/device/silicon_creator/mask_rom/mask_rom.c

```diff
@@ -111,20 +113,17 @@ static rom_error_t mask_rom_init(void) {
   // Initialize in-memory copy of the ePMP register configuration.
   mask_rom_epmp_state_init(&epmp, lc_state);
 
-  // Initialize the retention SRAM at power-on.
-  //
-  // The reset reason is treated as power-on if the POR bit is asserted
-  // regardless of whether or not any other reset reason bits are set. This is
-  // because we cannot guarantee that the retention SRAM was fully initialized
-  // if the device was reset before the POR bit was cleared.
-  //
-  // TODO(lowRISC/opentitan#7887): once the retention SRAM is initialized the
-  // reset reason should probably be saved into either main SRAM or the
-  // retention SRAM and the reset reason register cleared.
+  // Initialize the retention RAM based on the reset reason and the OTP value.
+  // Note: Retention RAM is always reset on PoR regardless of the OTP value.
   uint32_t reset_reasons = rstmgr_reason_get();
-  if (bitfield_bit32_read(reset_reasons, kRstmgrReasonPowerOn)) {
+  uint32_t reset_mask = (1 << kRstmgrReasonPowerOn)
+      | otp_read32(OTP_CTRL_PARAM_CREATOR_SW_CFG_RET_RAM_RESET_MASK_OFFSET);
+  if ((reset_reasons & reset_mask) != 0) {
```

Comment:
just to make sure i understand correctly...if ever we see a reset reason with multiple bits set, as long as ANY bit is set, we will still wipe out retention ram.  That's the intended behavior right?

Reply:
Not exactly: Retention RAM will be unconditionally initialized on PoR, for other reset reasons it depends on the OTP value.

Reply:
right sorry, i meant the latter case.
So let's say if the OTP sanctioned a particular reset reason, but the actual reset reason had multiple bits set.
As long as the bit we care about matches, we clear right? I think that's the right behavior, but i wanted to double check.

Reply:
Thanks for the clarification. You're right, I also think that's the right (and intended) behavior. Since bits are sticky, we can't be sure if ret_ram was initialized before the additional reset reasons occurred, so as long as one of the bits that we care about matches, we clear.

Reply:
sounds good, thanks for clarifying.